### PR TITLE
chore: add changelog for 13.8.0 release

### PR DESCRIPTION
**Pull request by AtofStryker**
Adds the changelog for Cypress 13.8.0

### CODE REVIEW

File: docs/guides/references/changelog.mdx [hunk 1]
```diff
@@ -2,6 +2,22 @@
...
+
+_Released 4/18/2024_
+
+**Features:**
+
+- Added support for `webpack-dev-server` `v5` to `@cypress/webpack-dev-server`. Addresses [#29305](https://github.com/cypress-io/cypress/issues/29305).
+
+**Bugfixes:**
+
+- We fixed a regression introduced in [`13.7.3`](https://docs.cypress.io/guides/references/changelog#13-7-3) where Cypress could hang handling long assertion messages. Fixes [#29350](https://github.com/cypress-io/cypress/issues/29350).
```
Comment (mschile) [inline docs/guides/references/changelog.mdx:15]: We don't typically use 'We'>

```suggestion
- Fixed a regression introduced in [`13.7.3`](https://docs.cypress.io/guides/references/changelog#13-7-3) where Cypress could hang handling long assertion messages. Fixes [#29350](https://github.com/cypress-io/cypress/issues/29350).
```

File: docs/guides/references/changelog.mdx [hunk 2]
```diff
@@ -2,6 +2,22 @@
...
+
+**Misc:**
+
+- We now capture the [Semaphore](https://semaphoreci.com/) CI provider's environment variable [`SEMAPHORE_GIT_PR_NUMBER`](https://docs.semaphoreci.com/ci-cd-environment/environment-variables/#semaphore_git_pr_number) to display the linked PR number in the Cloud. Addressed in [#29314](https://github.com/cypress-io/cypress/pull/29314).
```
Comment (mschile) [inline docs/guides/references/changelog.mdx:19]:
```suggestion
- The [Semaphore](https://semaphoreci.com/) CI provider's environment variable [`SEMAPHORE_GIT_PR_NUMBER`](https://docs.semaphoreci.com/ci-cd-environment/environment-variables/#semaphore_git_pr_number) is now captured to display the linked PR number in the Cloud. Addressed in [#29314](https://github.com/cypress-io/cypress/pull/29314).
```

**Review by mschile — CHANGES_REQUESTED** [review]
Suggested a few wording updates.

**Review by ryanthemanuel — APPROVED** [review]

File: docs/guides/references/changelog.mdx [hunk 3]
```diff
@@ -2,6 +2,22 @@
...
+
+- Added support for `webpack-dev-server` `v5` to `@cypress/webpack-dev-server`. Addresses [#29305](https://github.com/cypress-io/cypress/issues/29305).
+
+**Bugfixes:**
+
+- Fixed a regression introduced in [`13.7.3`](https://docs.cypress.io/guides/references/changelog#13-7-3) where Cypress could hang handling long assertion messages. Fixes [#29350](https://github.com/cypress-io/cypress/issues/29350).
+
+**Misc:**
+
+- The [Semaphore](https://semaphoreci.com/) CI provider's environment variable [`SEMAPHORE_GIT_PR_NUMBER`](https://docs.semaphoreci.com/ci-cd-environment/environment-variables/#semaphore_git_pr_number) is now captured to display the linked PR number in the Cloud. Addressed in [#29314](https://github.com/cypress-io/cypress/pull/29314).
```
Comment (mschile) [inline docs/guides/references/changelog.mdx:19]:
```suggestion
- The [`SEMAPHORE_GIT_PR_NUMBER`](https://docs.semaphoreci.com/ci-cd-environment/environment-variables/#semaphore_git_pr_number) environment variable from [Semaphore](https://semaphoreci.com/) CI is now captured to display the linked PR number in the Cloud. Addressed in [#29314](https://github.com/cypress-io/cypress/pull/29314).
```

**Review by mschile — APPROVED** [review]

**Review by jennifer-shehane — APPROVED** [review]

**Pull request merged**